plugins/probabilizer.cc

```diff
@@ -195,85 +172,12 @@ run(const ProcedureRunConfig & run,
                    runProcConf.trainingData.stm->offset,
                    runProcConf.trainingData.stm->limit);
 
-
-    int nx = numRows;
-
-    /* Convert to the correct data structures. */
-
-    boost::multi_array<double, 2> outputs(boost::extents[2][nx]);  // value, bias
-    distribution<double> correct(nx, 0.0);
-    distribution<double> weights(nx, 1.0);
-
-    size_t numTrue = 0;
-
-    for (unsigned x = 0;  x < nx;  ++x) {
-        float score, label, weight;
-        std::tie(std::ignore, score, label, weight) = fvs[x];
-
-        DEBUG_MSG(logger) 
-            << "x = " << x << " score = " << score << " label = " << label
-            << " weight = " << weight;
-
-        outputs[0][x] = score;
-        outputs[1][x] = 1.0;
-        correct[x]    = label;
-        weights[x]    = weight;
-        numTrue      += label == 1;
-    }
-
-#if 0
-    filter_ostream out("prob-in.txt");
-
-    for (unsigned i = 0;  i < nx;  ++i) {
-        out << MLDB::format("%.15f %.16f %d\n",
-                          outputs[0][i],
-                          outputs[1][i],
-                          correct[i]);
-
-    }
-#endif
-
-    double trueOneRate = correct.dotprod(weights) / weights.total();
-    double trueZeroRate = 1.0 - trueOneRate;
-
-    double numExamples = nx;
-    double sampleOneRate = numTrue / numExamples;
-    double sampleZeroRate = 1.0 - sampleOneRate;
-
-    INFO_MSG(logger) << "trueOneRate = " << trueOneRate
-                     << " trueZeroRate = " << trueZeroRate
-                     << " sampleOneRate = " << sampleOneRate
-                     << " sampleZeroRate = " << sampleZeroRate;
-
-    auto link = runProcConf.link;
-
-    ML::Ridge_Regressor regressor;
-    distribution<double> probParams
-        = ML::run_irls(correct, outputs, weights, link, regressor);
-
-    INFO_MSG(logger) << "probParams = " << probParams;
-
-    // http://gking.harvard.edu/files/0s.pdf, section 4.2
-    // Logistic Regression in Rare Events Data (Gary King and Langche Zeng)
-
-    double correction = -log((1 - trueOneRate) / trueOneRate
-                             * sampleOneRate / (1 - sampleOneRate));
-
-    DEBUG_MSG(logger) << "paramBefore = " << probParams[1];
-    DEBUG_MSG(logger) << "correction = " << correction;
-
-    probParams[1] += correction;
-
-    INFO_MSG(logger) << "paramAfter = " << probParams[1];
-
-    ProbabilizerRepr repr;
-    repr.style = "glz";
-    repr.link = link;
-    repr.params = probParams;
+    ProbabilizerModel probabilizer(logger);
+    probabilizer.train(fvs, runProcConf.link);
 
     if (!runProcConf.modelFileUrl.toString().empty()) {
         filter_ostream stream(runProcConf.modelFileUrl);
-        stream << jsonEncode(repr);
+        stream << jsonEncode(probabilizer);
```

Comment:
Are you 100% sure that the serialization format is exactly compatible with what was there before?